grai-integrations/source-bigquery/src/grai_source_bigquery/loader.py

```diff
@@ -269,3 +470,21 @@ def get_nodes_and_edges(self) -> Tuple[List[BigqueryNode], List[Edge]]:
         nodes = self.get_nodes()
         edges = self.get_edges()
         return nodes, edges
+
+    def get_nodes_and_edges(self) -> Tuple[List[BigqueryNode], List[Edge]]:
+        datasets = [self.dataset] if isinstance(self.dataset, str) else self.dataset
+
+        nodes = []
+        edges = []
+
+        for dataset in datasets:
+            self.dataset = dataset
```

Comment:
I'm a little scared of this sort of side-effecting particularly since it means get_nodes_and_edges isn't idempotent (self.dataset will be the last value in the list but you've lost the original reference, that's fixable but still sort of wonky.

I'd at least reset self.dataset to it's initial value but really it'd probably be better to move away from a cached_property to an lru_cache on the get methods and pass in a dataset value.

Reply:
I agree this is messy, I shall look into an `lru_cache`

Reply:
If nothing else I should use a different variable name